pkg/bisect/bisect.go

```diff
@@ -286,6 +288,28 @@ func (env *env) bisect() (*Result, error) {
 	return res, nil
 }
 
+func (env *env) testOriginalCommit() (*testResult, error) {
+	var testRes *testResult
+	var err error
+	for {
+		testRes, err = env.test()
+		if err != nil {
+			return nil, err
+		} else if testRes.verdict == vcs.BisectBad {
+			break
+		}
+
+		// Recheck original commit without ccache. We are disabling CONFIG_HAVE_GCC_PLUGINS
+		// and original crash might be dependent on that
+		if !env.disableCCache {
```

Comment:
This looks somewhat brute-forced for your current environment.
This is linux-specific, other OSes don't have CONFIG_HAVE_GCC_PLUGINS. And this OS-independent code.
A linux config may not contain (we don't have it on syzbot).
And ccache may not be even enabled in the first place.

Perhaps we could explicitly check what the comment talks about -- create BisectEnv with/without ccache and check if there is any difference.

Reply:
Or on second though:
1. Maybe it's not important enough and we just always do bisection with CONFIG_HAVE_GCC_PLUGINS disabled if ccache is enabled?

2. Or maybe we always generate `scripts/gcc-plugins/randomize_layout_seed.h` as the doc suggests and then we don't need to disable the plugin? It may be simple enough and will also increase reproducible without ccache and just make all other builds more reproducible.
If there are no hidden pitfalls, this looks like the best option.

Reply:
Fixed this as suggested in option 2.